xarray/core/dataarray.py

```diff
@@ -194,13 +192,14 @@ def __init__(self, data, coords=None, dims=None, name=None,
         attrs : dict_like or None, optional
             Attributes to assign to the new instance. By default, an empty
             attribute dictionary is initialized.
-        encoding : dict_like or None, optional
-            Dictionary specifying how to encode this array's data into a
-            serialized format like netCDF4. Currently used keys (for netCDF)
-            include '_FillValue', 'scale_factor', 'add_offset', 'dtype',
-            'units' and 'calendar' (the later two only for datetime arrays).
-            Unrecognized keys are ignored.
+        encoding : deprecated
         """
+
+        if encoding is not None:
+            warnings.warn(
+                'The `encoding` argument to `DataArray` is deprecated. '
+                'Instead, specify the encoding when writing to disk.',
+                FutureWarning)
```

Comment:
Be sure to specify `stacklevel` here (e.g., `stacklevel=2`) so users get a warning at the appropriate place in their code.